### PR TITLE
Also check `printf` and `Translations.HasKey` in translation key linter

### PR DESCRIPTION
**Pull request by pkong-ds**
ref DEV-1944

## Blocked by
- #4723

### CODE REVIEW

**Comment by pkong-ds** [comment]
Ready for review, thanks

File: devtools/gotemplatelinter/translation_key_rule_special_cases.go [hunk 1]
```diff
@@ -7,13 +7,29 @@ var allowedLegitTranslationKeys []string = []string{
...
 
+var printfLegitTranslationKeys []string = []string{
+	"territory-%s",
+	"language-%s",
+}
+var printfFalsePositives []string = []string{
+	"%s %s",
+	"radio-%s-%s",
+	"%s-icon",
+}
```
Comment (louischan-oursky) [inline devtools/gotemplatelinter/translation_key_rule_special_cases.go:18]: Remove printfFalsePositives.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by pkong-ds** [comment]
Updated, thanks!

**Comment by pkong-ds** [comment]
Thanks, did you change something? 🙏

**Comment by louischan-oursky** [comment]
I rebased and run `make lint-translation-keys` to resolve the conflicts for you.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**